02-transport-protocol.md

```diff
@@ -0,0 +1,60 @@
+# BOTD #2: Transport Protocol
+
+
+## Overview
+
+All communications between two *peers* SHOULD be encrypted in order to provide confidentiality for all transcripts and is authenticated in order to avoid malicious interference. 
+
+
+
+### Transport announcement
+
+Each *peer* MUST decides and MUST anounces which transport protocol wants to use for the connection, one or more combined between:
+
+
+* Clear text `insecure`
+* Onion service `onion`
+* Server-side TLS `TLS`
+* Mutual TLS `mTLS`
+* Noise_XK_secp256k1_ChaChaPoly_SHA256 `noise`
+
+
+Each peer SHOULD be reachable on the default TCP port **9945** and, if so, MUST expose then a plaintext HTTP/2 endpoint that announces the chosen transports responding with an `AvailableTransport` protobuf message, setting the response as `ContentType` as `application/protobuf`
```

Comment:
```suggestion
Each peer SHOULD be reachable on the default TCP port **9945** and, if so, MUST expose then a plaintext HTTP/2 endpoint that announces the chosen transports responding with an `AvailableTransport` protobuf message, setting the response `ContentType` as `application/protobuf`
```

>setting the response `ContentType` as `application/protobuf`

What are you trying to say with this? Why are you talking of ContentType that reminds to HTTP/1 request/response headers?

Reply:
Well HTTP/2 has content type same as 1 :D What I'm trying to say is that we are not forcing here to be gRPC, but plain http/2 with protobuf message as response.